sched/pthread/pthread_create.c

```diff
@@ -301,11 +303,16 @@ int pthread_create(FAR pthread_t *thread, FAR const pthread_attr_t *attr,
       goto errout_with_join;
     }
 
-#ifndef CONFIG_BUILD_KERNEL
-  /* Save the allocated task data in TLS */
+  /* Initialize thread local storage */
 
-  tls_set_taskdata(&ptcb->cmn);
-#endif
+  info = up_stack_frame(&ptcb->cmn, sizeof(struct tls_info_s));
+  if (info == NULL)
+    {
+      errcode = ENOMEM;
+      goto errout_with_join;
+    }
+
```

Comment:
I still don't understand this.  The high performance access to aligned stacks requires that the TLS data lie at the stack allocation access.  For a push down stack, up_stack_frame() allocates memory downward from the highest stack addresses.